dask/array/creation.py

```diff
@@ -12,8 +12,162 @@
 from ..base import tokenize
 from ..utils import ignoring
 from . import chunk
-from .core import Array, normalize_chunks, stack, concatenate
-from .wrap import empty
+from .core import Array, asarray, normalize_chunks, stack, concatenate
+from .wrap import empty, ones, zeros, full
+
+
+def empty_like(a, dtype=None, chunks=None):
+    """
+    Return a new array with the same shape and type as a given array.
+
+    Parameters
+    ----------
+    a : array_like
+        The shape and data-type of `a` define these same attributes of the
+        returned array.
+    dtype : data-type, optional
+        Overrides the data type of the result.
+    chunks : sequence of ints
+        The number of samples on each block. Note that the last block will have
+        fewer samples if ``len(array) % chunks != 0``.
+
+    Returns
+    -------
+    out : ndarray
+        Array of uninitialized (arbitrary) data with the same
+        shape and type as `a`.
+
+    See Also
+    --------
+    ones_like : Return an array of ones with shape and type of input.
+    zeros_like : Return an array of zeros with shape and type of input.
+    empty : Return a new uninitialized array.
+    ones : Return a new array setting values to one.
+    zeros : Return a new array setting values to zero.
+
+    Notes
+    -----
+    This function does *not* initialize the returned array; to do that use
+    `zeros_like` or `ones_like` instead.  It may be marginally faster than
+    the functions that do set the array values.
+    """
+
+    a = asarray(a)
```

Comment:
This means that we always have some kind of chunking. If not specified with `a` or the `chunks` argument, this ends up being the same as `a.shape`. This seems like reasonable behavior though we could raise for unspecified chunks instead.

Reply:
Using the input's chunks if not specified seems good to me